service/api/src/test/java/org/eclipse/kapua/KapuaRuntimeExceptionTest.java

```diff
@@ -0,0 +1,272 @@
+            }
+        }
+    }
+
+    @Test
+    public void kapuaRuntimeExceptionKapuaErrorCodeThrowableNullObjectParametesTest() {
+        for (Throwable throwable : throwables) {
+            KapuaException kapuaRuntimeException = new KapuaException(KapuaErrorCodes.ENTITY_NOT_FOUND, throwable, null);
+            assertEquals("Expected and actual values should be the same.", KapuaErrorCodes.ENTITY_NOT_FOUND, kapuaRuntimeException.getCode());
+            assertEquals("Expected and actual values should be the same.", "The entity of type {0} with id/name {1} was not found.", kapuaRuntimeException.getMessage());
+            assertEquals("Expected and actual values should be the same.", "The entity of type {0} with id/name {1} was not found.", kapuaRuntimeException.getLocalizedMessage());
+            assertEquals("Expected and actual values should be the same.", expectedErrorMessage, kapuaRuntimeException.getKapuaErrorMessagesBundle());
+            assertEquals("Expected and actual values should be the same.", throwable, kapuaRuntimeException.getCause());
+        }
+    }
+
+    @Test
+    public void internalErrorCauseMessageTest() {
+        String[] messages = {"Message", null};
+
+        for (Throwable throwable : throwables) {
+            for (String msg : messages) {
+                assertThat("Instance of KapuaRuntimeException expected.", KapuaRuntimeException.internalError(throwable, msg), IsInstanceOf.instanceOf(KapuaRuntimeException.class));
+                assertEquals("Expected and actual values should be the same.", new KapuaRuntimeException(KapuaErrorCodes.INTERNAL_ERROR, throwable, msg).toString(), KapuaRuntimeException.internalError(throwable, msg).toString());
+                assertEquals("Expected and actual values should be the same.", "An internal error occurred: " + msg + ".", KapuaRuntimeException.internalError(throwable, msg).getMessage());
+                assertEquals("Expected and actual values should be the same.", throwable, KapuaRuntimeException.internalError(throwable, msg).getCause());
+                assertEquals("Expected and actual values should be the same.", KapuaErrorCodes.INTERNAL_ERROR, KapuaRuntimeException.internalError(throwable, msg).getCode());
+                assertEquals("Expected and actual values should be the same.", expectedErrorMessage, KapuaRuntimeException.internalError(throwable, msg).getKapuaErrorMessagesBundle());
+            }
+        }
+    }
+
+    @Test
+    public void internalErrorCauseTest() {
+        String message = "Message";
+        Throwable[] throwables = {new Throwable(message), new Throwable()};
+        Throwable nullThrowable = null;
+        String[] expectedMessage = {"An internal error occurred: " + message + ".", "An internal error occurred: " + throwables[1] + "."};
+        String[] arguments = {"Message", "java.lang.Throwable"};
+
+        for (int i = 0; i < throwables.length; i++) {
```

Comment:
Please use enhanced for loop: for(Throwable thr:throwables)

Reply:
I used index i to get values from different arrays in the same position